### PR TITLE
Update CONTRIBUTING.md (and related stuff)

### PR DESCRIPTION
**Pull request by gildegoma**
Motivated by https://twitter.com/mitchellh/status/996381507756294144

@mitchellh @chrisroberts @briancain this is an attempt to cherry-pick some good parts from Packer and Terraform contributing guide lines, trying to adapt them to my vision/understanding on how Vagrant development dynamics work. I hope it fits with your perspectives as well. But if it is not the case, please close and start a new topic branch, I won't be offended at all ;-)

For english-quality and project-ownership reason, I'd appreciate if Chris or Brian could take the lead to polish and adapt the changes proposed here... Many thanks in advance!

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -1,41 +1,68 @@
...
+4. The issue is addressed in a pull request or commit. The issue will be referenced in the commit message so that the code that fixes it is clearly linked.
+5. The issue is closed. Sometimes, valid issues will be closed to keep the issue tracker clean. The issue is still indexed and available for future viewers, or can be re-opened if necessary.
+
+## Pull Requests
+
+Thank you for contributing! Here you'll find information on what to include in your Pull Request (“PR” for short) to ensure it is reviewed quickly, and possibly accepted.
+
+Before starting work on a new feature or anything besides a minor bug fix, it is highly recommended to first initiate a discussion with the Vagrant community (either via a GitHub issue, the [mailing list], IRC freenode `#vagrant` or [Gitter](https://gitter.im/mitchellh/vagrant)). This will save you from wasting days implementing a feature that could be rejected in the end.
```
Comment (briancain) [inline .github/CONTRIBUTING.md]: Is the `[mailing list]` block here missing the URL?

**Comment by briancain** [comment]
Thanks @gildegoma - just a single comment, otherwise looks good to me! Thanks for updating this for us.

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (gildegoma) [reply .github/CONTRIBUTING.md]: @briancain Thanks for the good catch! 

:bookmark_tabs: I actually noticed that I was about to forget the same URL at [line 7](https://github.com/hashicorp/vagrant/pull/9841/commits/4471bc48fb73c7d4774d7006a979dc97701147c3#diff-8cc7b2b0d78dd2501610391c086a8516R7) and fixed it before pushing, without thinking it was mentioned twice in the same document :laughing:

**Comment by gildegoma** [comment]
@briancain Thanks for reviewing it! I am happy if you agree!
I defer to you or @chrisroberts to the merge (feel free to rebase/fixup/squash at will).

**Comment by gildegoma** [comment]
😊🎉

**Pull request merged**